sd-rsyslog

```diff
@@ -0,0 +1,193 @@
+'omprog' action like the following:
+    action(type="omprog"
+        binary="/usr/bin/myplugin.py"
+        output="/var/log/myplugin.log"
+        confirmMessages="on"
+        ...)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+     -or-
+     see COPYING.ASL20 in the source distribution
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import sys
+import os
+import logging
+import configparser
+from subprocess import Popen, PIPE
+
+# Global definitions specific to your plugin
+process = None
+
+class RecoverableError(Exception):
+    """An error that has caused the processing of the current message to
+    fail, but does not require restarting the plugin.
+
+    An example of such an error would be a temporary loss of connection to
+    a database or a server. If such an error occurs in the onMessage function,
+    your plugin should wrap it in a RecoverableError before raising it.
+    For example:
+
+        try:
+            # code that connects to a database
+        except DbConnectionError as e:
+            raise RecoverableError from e
+
+    Recoverable errors will cause the 'omprog' action to be temporarily
+    suspended by rsyslog, during a period that can be configured using the
+    "action.resumeInterval" action parameter. When the action is resumed,
+    rsyslog will resend the failed message to your plugin.
+    """
+
+
+def onInit():
+    """Do everything that is needed to initialize processing (e.g. open files,
+    create handles, connect to systems...).
+    """
+    # Apart from processing the logs received from rsyslog, you want your plugin
+    # to be able to report its own logs in some way. This will facilitate
+    # diagnosing problems and debugging your code. Here we set up the standard
+    # Python logging system to output the logs to stderr. In the rsyslog
+    # configuration, you can configure the 'omprog' action to capture the stderr
+    # of your plugin by specifying the action's "output" parameter.
+    logging.basicConfig(stream=sys.stderr,
+                        level=logging.WARNING,
+                        format='%(asctime)s %(levelname)s %(message)s')
+
+    # This is an example of a debug log. (Note that for debug logs to be
+    # emitted you must set 'level' to logging.DEBUG above.)
+    logging.debug("onInit called")
+
+
+    global process
+    if not os.path.exists("/etc/sd-rsyslog.conf"):
+        print("Please create the configuration file at /etc/sd-rsyslog.conf", file=sys.stderr)
+        sys.exit(1)
+    config = configparser.ConfigParser()
```

Comment:
Yes, I tried this path first, but, this means we will modifying the rsyslog output plugin file while installing, I chose the path of a config file that sounded easier or more maintainable to me.

Reply:
We can either configure this output plugin, or a simpler configuration file, I chose to do that later as it will easier to maintain and understand in the future.

Reply:
Yeah, you're right. Everyone knows INI-style config files, and rsyslog is relatively obscure.